arangod/Graph/Providers/SingleServerProvider.cpp

```diff
@@ -141,39 +123,73 @@ auto SingleServerProvider<Step>::expand(
     std::function<void(Step)> const& callback) -> void {
   TRI_ASSERT(!step.isLooseEnd());
   auto const& vertex = step.getVertex();
-  TRI_ASSERT(_cursor != nullptr);
+
   LOG_TOPIC("c9169", TRACE, Logger::GRAPHS)
       << "<SingleServerProvider> Expanding " << vertex.getID();
-  _cursor->rearm(vertex.getID(), step.getDepth(), _stats);
-  ++_rearmed;
-  _cursor->readAll(
-      *this, _stats, step.getDepth(),
-      [&](EdgeDocumentToken&& eid, VPackSlice edge, size_t cursorID) -> void {
-        ++_readSomething;
-        VertexType id = _cache.persistString(([&]() -> auto {
-          if (edge.isString()) {
-            return VertexType(edge);
-          } else {
-            VertexType other(
-                transaction::helpers::extractFromFromDocument(edge));
-            if (other == vertex.getID()) {  // TODO: Check getId - discuss
-              other =
-                  VertexType(transaction::helpers::extractToFromDocument(edge));
-            }
-            return other;
-          }
-        })());
-        // TODO: Adjust log output
-        LOG_TOPIC("c9168", TRACE, Logger::GRAPHS)
-            << "<SingleServerProvider> Neighbor of " << vertex.getID() << " -> "
-            << id;
-
-        callback(Step{id, std::move(eid), previous, step.getDepth() + 1,
-                      _opts.weightEdge(step.getWeight(), edge), cursorID});
-        // TODO [GraphRefactor]: Why is cursorID set, but never used?
-        // Note: There is one implementation that used, it, but there is a high
-        // probability we do not need it anymore after refactoring is complete.
-      });
+
+  std::vector<ExpansionInfo>* neighbours;
+  std::vector<ExpansionInfo> newNeighbours;  // only used if not in cache
```

Comment:
This is only used in the else case starting in line 140, so let's move this declaration in there.

Reply:
Actually not, if the cache is not used, we collect the data there and then point the pointer `neighbours` to it, and then it is used further down. So it needs to stay in this scope.

Reply:
Done with the move to `shared_ptr`s below.